File: 2225-find-players-with-zero-or-one-losses/2225-find-players-with-zero-or-one-losses.py
class Solution(object):
    def findWinners(self, matches):
        winners = []
        loosers1 = []
        res = []

        score_map = defaultdict(lambda: [0, 0])

        for pair in matches:
            winner, looser = pair[0], pair[1]
            score_map[winner][0] += 1
            score_map[looser][1] += 1

        for key, value in score_map.items():
            loosing_score = value[1]
            if loosing_score == 0:
                winners.append(key)
            if loosing_score == 1:
                loosers1.append(key)

        winners.sort()
        loosers1.sort()
        res.append(winners)
        res.append(loosers1)
        return res
        

File: 2225-find-players-with-zero-or-one-losses/2225-find-players-with-zero-or-one-losses.java
class Solution {
    public List<List<Integer>> findWinners(int[][] matches) {
        List<Integer> winners = new ArrayList<>();
        List<Integer> loosers1 = new ArrayList<>();
        List<List<Integer>> res = new ArrayList<>();

       Map<Integer,int[]> map = new TreeMap<>();

       for(int[] pair : matches){
           int winner = pair[0], looser = pair[1];
           if(map.containsKey(winner)) {
                map.get(winner)[0]++;
           }else map.put(winner,new int[]{1,0});

           if(map.containsKey(looser)){
               map.get(looser)[1]++;
           }else map.put(looser,new int[]{0,1});
       }

        for (var entry : map.entrySet()){
            int loosingScore = entry.getValue()[1];
            if(loosingScore == 0) winners.add(entry.getKey());
            if(loosingScore == 1) loosers1.add(entry.getKey());
        }

        //  Collections.sort(winners);
        // Collections.sort(loosers1);
        res.add(winners);
        res.add(loosers1);
        return res;
    }
}